src/compute/src/logging/compute.rs

```diff
@@ -231,17 +231,17 @@ pub fn construct<A: Allocate>(
         });
         let peek_current = peek.as_collection().map(move |datum| {
             Row::pack_slice(&[
-                Datum::Uuid(datum.peek.uuid),
-                Datum::UInt64(u64::cast_from(datum.worker)),
-                Datum::String(&datum.peek.id.to_string()),
-                Datum::MzTimestamp(datum.peek.time),
+                Datum::Uuid(datum.uuid),
+                Datum::UInt64(u64::cast_from(worker)),
+                Datum::String(&datum.id.to_string()),
+                Datum::MzTimestamp(datum.time),
             ])
         });
         let peek_duration = peek_duration
             .as_collection()
-            .arrange_named::<RowSpine<_, _, _, _>>("Arranged timely peek_duration")
+            .arrange_core::<_, RowSpine<_, _, _, _>>(Pipeline, "Arranged timely peek_duration")
```

Comment:
Instead of `Pipeline`, include the worker and use a defensive exchange, like it's done in Timely logging:
```
            .arrange_core::<_, RowSpine<_, _, _, _>>(
                Exchange::new(|(((_, w), ()), _, _)| u64::cast_from(*w)),
                "PreArrange Timely duration",
            )
```
This will make sure data is arranged correctly, without incurring a significant overhead.